changelog/cwd-fix.dd

```diff
@@ -0,0 +1,14 @@
+Fixed regression with `dub --root=x` from last release
+
+Since last release builds via the DUB API rely on the package path or an
+explicitly set working directory, instead of the apps working directory. However
+in the `dub` command line tool the previous behavior was when using
+`dub run --root=path`, the built app that is being run would be run from dubs
+current cwd, unless `workingDirectory` is specified in the recipe. Now since the
+last release this changed to the package directory (e.g. like defaulting
+`"workingDirectory": "."` in dub.json), which is considered a breaking change /
+regression.
+
+So this release reverts back the behavior that was accidentally changed, so that
+the `dub` CLI working directory is always being used as program cwd when no
+working directory is set in the package recipe.
```

Comment:
```
Reverted v1.31.0 working directory change when using `dub run --root=<value>`

dub < v1.31.0 would run applications in the working directory it was invoked in, ignoring the `--root` argument.
In v1.31.0, `dub` started to respect the `--root` argument and run programs in the requested directory.
While desirable, this change was not intended, and has now been reverted. It will be re-introduced as part of a major version upgrade.
```

Let me know if that works for you. I tried to rephrase the text keeping the important informations you already put in, but doing it more succinctly.

Reply:
I have adjusted it to mostly that, but omitted that it would be re-introduced. I think the workingDirectory: "." setting is a good universal solution to cover all use cases. (those who want the dub cwd as part of some CI run and those who want to control it to their package)